compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -368,11 +391,103 @@ impl<'tcx> SizeSkeleton<'tcx> {
             (SizeSkeleton::Pointer { tail: a, .. }, SizeSkeleton::Pointer { tail: b, .. }) => {
                 a == b
             }
+            // constants are always pre-normalized into a canonical form so this
+            // only needs to check if their pointers are identical.
+            (SizeSkeleton::Generic(a), SizeSkeleton::Generic(b)) => a == b,
             _ => false,
         }
     }
 }
 
+fn combine_sorted_expr<'tcx>(
```

Comment:
This doesn't imply "multiplication" to me, am I missing something?

Reply:
I was thinking about almost like a merge in a merge-sort, I can rename it to better fit but since it was a private fn I didn't think too much about it